File: front-eccommerce/src/app/adpaters/pedidoAdapter.ts
import {PedidoDto} from "../entity/dto/pedidoDto";
import {Pedido} from "../entity/pedido";

export class PedidoAdapter {

  pedidoAdapter(pedido: Pedido): PedidoDto {
    let p: PedidoDto = new PedidoDto();
    p.id = pedido.idUsuario.idUsuario;
    p.realizado = pedido.realizado
    p.precioTotal = pedido.precioTotal;
    p.idUsuario = pedido.idUsuario;
    p.idPago = pedido.idPago;
    return p;
  }
}

File: sota-ecommerce/src/main/java/com/sota/net/adapters/PedidoAdapter.java
package com.sota.net.adapters;

import com.sota.net.entity.Pedido;
import com.sota.net.entity.Usuario;
import com.sota.net.entity.dto.PedidoCreadoDto;

import java.util.ArrayList;

public class PedidoAdapter {
    public static Pedido getPedido(PedidoCreadoDto pedidoDto, Usuario u) {
        Pedido pedido = new Pedido();

        pedido.setIdUsuario(u);

        pedido.setPrecioTotal(pedidoDto.getPrecioTotal());
        pedido.setIdPago(pedidoDto.getIdPago());
        pedido.setPedidoProducto(new ArrayList<>());
        return pedido;
    }
}
